### PR TITLE
Add GN second kind element

### PR DESCRIPTION
**Pull request by mscroggs**
Progress towards #344

### CODE REVIEW

File: elements/bernardi-guzman-neilan-raugel.def [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Bernardi-Guzman-Neilan-Raugel
```
Comment (mscroggs) [inline elements/bernardi-guzman-neilan-raugel.def:1]: @pbrubeck What do you think of this name for the "Guzman-Neilan" element currently in Symfem? Once we agree on a name, I'll update the name used in Symfem itself to match

File: elements/guzman-neilan.def [hunk 2]
```diff
@@ -25,10 +25,7 @@ dofs:
     - normal integral moments with (lagrange,0)
 mapping: contravariant Piola
 sobolev: H1
-notes:
-  - This element is a modification of the [Bernardi&ndash;Raugel](element::bernardi-raugel) element with the facet bubbles modified to be divergence free.
```
Comment (mscroggs) [inline elements/guzman-neilan.def:29]: I think this not is no longer relevant to the new Guzman-Neilan?

Comment (pbrubeck) [reply elements/guzman-neilan.def:29]: This note was not relevant before, now it is.

File: elements/bernardi-guzman-neilan-raugel.def [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - vector
+  - Hdiv
+  - macro
+reference-elements:
+  - triangle
+  - tetrahedron
+dofs:
+  edges: (if \(k>1\)) point evaluations in tangential directions at midpoints
+  facets:
+    - point evaluations in normal directions at vertices
```
Comment (pbrubeck) [inline elements/bernardi-guzman-neilan-raugel.def:24]: The element is always H^1 conforming, point evaluations at vertices most be associated to the vertices.

File: elements/bernardi-guzman-neilan-raugel.def [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    - normal integral moments with (lagrange,0)
+    - (if \(k>1\)) point evaluations in normal directions at midpoints of edges
+  cell:
+    - point evaluations at the midpoint
+mapping: contravariant Piola
+sobolev: H(div)
+notes:
+  - This element is a modification of the [Bernardi&ndash;Raugel](element::bernardi-raugel) element with the facet bubbles modified to be divergence free.
```
Comment (pbrubeck) [inline elements/bernardi-guzman-neilan-raugel.def]: This element is BR with modified bubbles, plus interior (macro) bubbles: the hat function associated to the barycenter for each coordinate direction.

File: elements/bernardi-guzman-neilan-raugel.def (hunk 1 above)
Comment (mscroggs) [reply elements/bernardi-guzman-neilan-raugel.def]: that's a better suggestion. The element implemented in FIAT is not "Guzman-Neilan (first kind)" with alternative name "Guzman-Neilan". The element currently implemented almost correctly in Symfem is "Guzman-Neilan (second kind)"

File: elements/guzman-neilan.def (hunk 2 above)
Comment (mscroggs) [reply elements/guzman-neilan.def:29]: You're right, I moved this note back to GN first kind

File: elements/guzman-neilan2.def [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    - k=2: 49
+categories:
+  - vector
+  - Hdiv
+  - macro
+reference-elements:
+  - triangle
+  - tetrahedron
+dofs:
+  vertices: point evaluations in normal directions at vertices
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:20]:
```suggestion
  vertices: point evaluations in coordinate directions
```

Comment (pbrubeck) [reply elements/guzman-neilan2.def]: Also there I left a typo here, but for the first kind

File: elements/guzman-neilan2.def [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  edges: (if \(k>1\)) point evaluations in tangential directions at midpoints
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:21]:
```suggestion
  edges: (if \(k>1\)) point evaluations in coordinate directions at midpoints
```

File: elements/guzman-neilan2.def [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  facets:
+    - normal integral moments with (lagrange,0)
+    - (if \(k>1\)) point evaluations in normal directions at midpoints of edges
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:24]:
```suggestion
```

File: elements/guzman-neilan2.def [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  cell:
+    - point evaluations at the midpoint
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:26]:
```suggestion
    - point evaluations in coordinate directions at the midpoint
```

File: elements/guzman-neilan2.def [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+mapping: contravariant Piola
+sobolev: H(div)
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:28]:
```suggestion
sobolev: H1
```

File: elements/guzman-neilan2.def [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+ndofs:
+  triangle:
+    formula: 11
+  tetrahedron:
+    formula:
+    - k=1: 19
+    - k=2: 49
+categories:
+  - vector
+  - Hdiv
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:14]:
```suggestion
```

File: elements/guzman-neilan2.def [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+reference-elements:
+  - triangle
+  - tetrahedron
+dofs:
+  vertices: point evaluations in coordinate directions
+  edges: (if \(k>1\)) point evaluations in coordinate directions at midpoints
+  facets:
+    - normal integral moments with (lagrange,0)
+  cell:
+    - point evaluations in coordinate directions at the midpoint
```
Comment (pbrubeck) [inline elements/guzman-neilan2.def:24]: I think for (k=2) this has 4 more point evaluations at each interior edge of the Alfeld split

Comment (mscroggs) [reply elements/guzman-neilan2.def:24]: I'm going to correct the implementation in Symfem, then I'll fix this

Comment (pbrubeck) [reply elements/guzman-neilan2.def:24]: For k=2 you need 30 additional dofs, which should come as 3 components on 10 edges = 6 exterior + 4 interior

File: elements/bernardi-guzman-neilan-raugel.def (hunk 1 above)
Comment (pbrubeck) [reply elements/bernardi-guzman-neilan-raugel.def]: The element in FIAT is Guzman-Neilan extended to (Pk + FacetBubble)^d, the first (dim+1)^2 dofs give you GN

**Pull request merged**